python/ray/autoscaler/_private/load_metrics.py

```diff
@@ -125,13 +125,17 @@ def get_resource_utilization(self):
         return self.dynamic_resources_by_ip
 
     def _get_resource_usage(self):
-        num_nodes = len(self.static_resources_by_ip)
+        num_nodes = 0
         nodes_used = 0.0
         num_nonidle = 0
         has_saturated_node = False
         resources_used = {}
         resources_total = {}
         for ip, max_resources in self.static_resources_by_ip.items():
+            # Nodes without resources don't count as nodes (e.g. unmanaged
+            # nodes)
+            if any(max_resources.values()):
+                num_nodes += 1
```

Comment:
I'm a bit confused what this is doing, it seems to increment num_nodes if any resource value is non-zero.

Doesn't this double count all nodes?

Reply:
Why would this double count the nodes? We go through each node in `static_resources` exactly once right?

Reply:
Got it. It didn't see num_nodes=0 in the earlier patch.